### PR TITLE
Serializing ints..

### PR DESCRIPTION
**Pull request by svogl**
Hi all,
I thought I'd try out the clone/pull-request feature of github - the code is equivalent to bug #69 on
http://ape-project.lighthouseapp.com/projects/39102/tickets/69-serializing-numbers-to-ints#ticket-69-1

Hope I don't bug you too  much ;)
Simon

### CODE REVIEW

**Comment by paraboul** [comment]
Thanks for that.

I'll check that ASAP

**Comment by svogl** [comment]
Fine - you might to look at my subsequent patch as well -> spidermonkey uses doubles for large ints, I'm trying to convert them back as well (this happend to the timestamps I transmit ;).

File: modules/libape-spidermonkey.c [hunk 1]
```diff
@@ -452,15 +452,32 @@ static json_item *jsobj_to_ape_json(JSContext *cx, JSObject *json_obj)
...
+						} else {
+							json_set_element_int(ape_json, di);
+						}
 					} else {
-						/* json_set_element_int(ape_json, (long int)dp); */
-						json_set_element_float(ape_json, dp);
+						jsdouble dp;
+						JS_ValueToNumber(cx, vp, &dp);
+						
+						long long ll = (long long)trunc(dp);
```
Comment (paraboul) [inline modules/libape-spidermonkey.c:467]: Can you try this instead ? It should be faster :)  :

jsint ip;
ip = dp;

if ((jsint)dp != ip) {

**Comment by paraboul** [comment]
Ok, it doesn't work on 32bit system.
The issue come from the JSON parser. Let me patch this.

**Comment by svogl** [comment]
you're right - we've got a 32bit machine here - tried the casting first which did not work of course so I searched for a different solution, but did not have too much time to ponder about it, so I thought clean solution goes over fast hack ;)

**Comment by paraboul** [comment]
Fixed : https://github.com/APE-Project/APE_Server/commit/320a5af6c732d131cbdd8a2e9b78c8abde70c573

Thanks ;)

**Pull request closed without merging**